File: src/communication/ComServer.java
package communication;

import java.io.IOException;
import java.net.ServerSocket;

import common.Main;

public class ComServer implements Runnable {
	
	private ServerSocket _SS;
	private Thread _t;
	
	public ComServer() 
	{
		try
		{
			_SS = new ServerSocket(Main.REALM_COM_PORT);
			_t = new Thread(this);
			_t.setDaemon(true);
			_t.start();
		}catch(IOException e)
		{
			System.out.println("COMServer: "+e.getMessage());
			Main.agregaralogdecom("COMServer: "+e.getMessage());
			Main.agregaralogdeerrores("COMServer: "+e.getMessage());
			Main.cerrarservidores();
		}
	}
	
	public void run() 
	{
		while(Main.isRunning)
		{
			try
			{
				new ComThread(_SS.accept());
			}catch(IOException e)
			{
				try {
					if(!_SS.isClosed()) _SS.close();
					} catch (IOException e1) {}
				System.out.println("COMServer Ejecutando: "+e.getMessage());
				Main.agregaralogdecom("COMServer Ejecutando: "+e.getMessage());
				Main.agregaralogdeerrores("COMServer Ejecutando: "+e.getMessage());
			}
		}
	}
	
	public void kickAll() {
		try {
			_SS.close();
		}catch(Exception e) {
			System.out.println("ComServerKickAll : "+e.getMessage());
			Main.agregaralogdecom("ComServerKickAll : "+e.getMessage());
			Main.agregaralogdeerrores("ComServerKickAll : "+e.getMessage());
		}
	}
	public Thread getThread() 
	{
		return _t;
	}
}

File: src/main/kotlin/communication/ComServer.kt
package communication

import common.Config
import java.io.IOException
import java.net.ServerSocket

class ComServer : Runnable {

    private var _SS: ServerSocket? = null
    private var thread: Thread? = null

    init {
        try {
            _SS = ServerSocket(Config.REALM_COM_PORT)
            thread = Thread(this)
            thread?.isDaemon = true
            thread?.start()
        } catch (e: IOException) {
            println("COMServer: " + e.message)
            Config.agregaralogdecom("COMServer: " + e.message)
            Config.agregaralogdeerrores("COMServer: " + e.message)
            Config.cerrarservidores()
        }
    }

    override fun run() {
        while (Config.isRunning) {
            try {
                ComThread(_SS!!.accept())
            } catch (e: IOException) {
                try {
                    if (!_SS!!.isClosed) _SS!!.close()
                } catch (e1: IOException) {
                }
                println("COMServer Ejecutando: " + e.message)
                Config.agregaralogdecom("COMServer Ejecutando: " + e.message)
                Config.agregaralogdeerrores("COMServer Ejecutando: " + e.message)
            }
        }
    }

    fun kickAll() {
        try {
            _SS!!.close()
        } catch (e: Exception) {
            println("ComServerKickAll : " + e.message)
            Config.agregaralogdecom("ComServerKickAll : " + e.message)
            Config.agregaralogdeerrores("ComServerKickAll : " + e.message)
        }
    }
}